tests/tracking/test_tracking.py

```diff
@@ -34,6 +31,33 @@ def test_create_experiment():
         tracking.set_tracking_uri(None)
 
 
+def test_set_experiment():
+    with pytest.raises(TypeError):
+        mlflow.set_experiment()
+
+    with pytest.raises(Exception):
+        mlflow.set_experiment(None)
+
+    with pytest.raises(Exception):
+        mlflow.set_experiment("")
+
+    assert not tracking.fluent._active_experiment_id
+
+    try:
+        with TempDir() as tracking_uri:
+            tracking.set_tracking_uri(tracking_uri.path())
+            name = "Some random experiment name %d" % random.randint(1, 1e6)
+            exp_id = mlflow.create_experiment(name)
+            mlflow.set_experiment(name)
+            assert mlflow.tracking.fluent._active_experiment_id == exp_id
+
+            another_name = "Another random name %d" % random.randint(1, 1e6)
```

Comment:
There's a small chance this will clash! I'd prefer not to use random at all in tests; just set them to deterministic names.

Reply:
There is no chance of conflict. Original experiment has `"Some random experiment name"` as prefix and next one has `"Another random name"`. Furthermore, this logged locally under new `TempDir`. 

But in this case, random suffixes are not buying us anything. I will make them deterministic and something same for experiment names.